linalg/src/arm64/arm64fp16/by_scalar.rs

```diff
@@ -34,8 +34,183 @@ ew_impl_wrap!(
     }
 );
 
+by_scalar_impl_wrap!(
+    f16,
+    arm64fp16_add_by_scalar_f16_32n,
+    32,
+    4,
+    f16,
+    fn run(buf: &mut [f16], s: f16) {
+        assert!(buf.len() % 16 == 0);
+        assert!(buf.len() > 0);
+        #[target_feature(enable = "fp16")]
+        unsafe fn run(buf: &mut[f16], s: f16) {
+            let len = buf.len();
+            let ptr = buf.as_ptr();
+            std::arch::asm!("
+            dup v0.8h, v0.h[0]
+            2:
+                ld1 {{v4.8h, v5.8h, v6.8h, v7.8h}}, [{ptr}]
```

Comment:
we need to give a shot at unrolling to amortize the loading latency, and prefetch too (but this is another task/PR)